### PR TITLE
PanelLibrary: changes casing of responses and adds meta property

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This PR changes the responses of the library panel api:
- adds a meta property with user names and canEdit flag (always true for now)
- changes casing so the api is more easily consumed by the frontend

**Which issue(s) this PR fixes**:
Relates #29610

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/librarypanels/database.go [hunk 1]
```diff
@@ -148,50 +166,109 @@ func (lps *LibraryPanelService) disconnectLibraryPanelsForDashboard(dashboardID
...
-	session.Table("library_panel")
-	session.Where("uid=? AND org_id=?", uid, orgID)
-	err := session.Find(&libraryPanels)
+func getLibraryPanel(session *sqlstore.DBSession, uid string, orgID int64) (LibraryPanelWithMeta, error) {
+	libraryPanels := make([]LibraryPanelWithMeta, 0)
+	sql := `SELECT
+				lp.id, lp.org_id, lp.folder_id, lp.uid, lp.name, lp.model, lp.created, lp.created_by, lp.updated, lp.updated_by
+				, 0 as can_edit
+				, (SELECT u.login from user u where u.id = lp.created_by and u.org_id = lp.org_id) as created_by_name
+				, (SELECT u.login from user u where u.id = lp.updated_by and u.org_id = lp.org_id) as updated_by_name
```
Comment (hugohaggmark) [inline pkg/services/librarypanels/database.go:175]: Is there a more better way to do this SELECT without sub selects? I'm a bit rusty here and appreciate any suggestions.

File: pkg/services/librarypanels/database.go [hunk 2]
```diff
@@ -148,50 +166,109 @@ func (lps *LibraryPanelService) disconnectLibraryPanelsForDashboard(dashboardID
...
 	orgID := c.SignedInUser.OrgId
-	libraryPanels := make([]LibraryPanel, 0)
+	libraryPanels := make([]LibraryPanelWithMeta, 0)
 	err := lps.SQLStore.WithDbSession(context.Background(), func(session *sqlstore.DBSession) error {
-		err := session.SQL("SELECT * FROM library_panel WHERE org_id=?", orgID).Find(&libraryPanels)
+		sql := `SELECT
+				lp.id, lp.org_id, lp.folder_id, lp.uid, lp.name, lp.model, lp.created, lp.created_by, lp.updated, lp.updated_by
+				, 0 as can_edit
+				, (SELECT u.login from user u where u.id = lp.created_by and u.org_id = lp.org_id) as created_by_name
+				, (SELECT u.login from user u where u.id = lp.updated_by and u.org_id = lp.org_id) as updated_by_name
```
Comment (hugohaggmark) [inline pkg/services/librarypanels/database.go:235]: Same comment as above, could this SELECT be done in a better way?

File: pkg/services/librarypanels/database.go (hunk 1 above)
Comment (torkelo) [reply pkg/services/librarypanels/database.go:175]: should be able to just do a join, important to get the user email so you can return the avatar image url as well, see https://github.com/grafana/grafana/blob/master/pkg/services/sqlstore/team.go#L409 for reference

Comment (hugohaggmark) [reply pkg/services/librarypanels/database.go:175]: Thank you, I'll look into this.

Comment (hugohaggmark) [reply pkg/services/librarypanels/database.go]: I couldn't figure out how to use `xorm` with two joins using the same table name so I wrote it like an SQL statement instead.

File: pkg/services/librarypanels/models.go [hunk 3]
```diff
@@ -22,6 +22,56 @@ type LibraryPanel struct {
...
+	CreatedByName  string
+	CreatedByEmail string
+	UpdatedByName  string
+	UpdatedByEmail string
+}
+
+// LibraryPanelDTO is the frontend DTO for library panels.
+type LibraryPanelDTO struct {
+	ID       int64               `json:"id"`
+	OrgID    int64               `json:"orgID"`
```
Comment (torkelo) [inline pkg/services/librarypanels/models.go:49]: should this be orgId (in the json)?

**Review by torkelo — APPROVED** [review]
Think this looks good, just one minor json prop name issue

File: pkg/services/librarypanels/models.go (hunk 3 above)
Comment (hugohaggmark) [reply pkg/services/librarypanels/models.go:49]: Yes, of course. Good catch!

**Comment by hugohaggmark** [comment]
Any last feedback from @kaydelaney, @aknuds1 or @kylebrandt before I merge this?

**Comment by hugohaggmark** [comment]
I'm merging this PR now. Please leave any additional comments and I'll fix them in a later PR @aknuds1 @kylebrandt. Thank you.

**Pull request merged**